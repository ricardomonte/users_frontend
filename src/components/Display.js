import React, { useEffect, useState } from 'react';
import SearchUser from './SearchUser';
import Card, { CardPrimaryContent } from "@material/react-card";
import {
  Headline6,
  Overline,
} from '@material/react-typography';
import '@material/react-card/dist/card.css';
import '@material/elevation/dist/mdc.elevation.css';
import '@material/react-typography/dist/typography.css';

const Display = ({users}) => {
  const [listUsers, setListUsers] = useState([]);
  const [filterUsers, setFilterUsers] = useState('');


  useEffect(() => {
    setListUsers(users.data)
  },[users]);

  const handleChange = (e) => {
    setFilterUsers(e.target.value)
  }

  const userFilter = () => listUsers.filter((user) =>
    user.attributes.name.toLowerCase().includes(filterUsers.toLowerCase())
    ||
    user.attributes.lastname.toLowerCase().includes(filterUsers.toLowerCase()))

  const filteredListUsers = filterUsers === '' ? listUsers : userFilter()

  return (
    <div style={{width: '40%' }}>
      <Card className='mdc-card demo-card demo-ui-control' style={{rowGap: '2em', padding: '2em'}}>        
      <SearchUser term={filterUsers} handleSearch={handleChange} />
        { filteredListUsers.map((user) => (
          <CardPrimaryContent className='mdc-card__primary-action mdc-elevation--z4'>
            <div>
              <Headline6>{`${user.attributes.name} ${user.attributes.lastname}`}</Headline6>
            </div> 
            <div>
              <Overline>{user.attributes.email}</Overline>
            </div> 
          </CardPrimaryContent>
        )) }
      </Card>
    </div>
  );
};

export default Display