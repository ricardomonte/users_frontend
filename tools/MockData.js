const MockData = {
  data: [
    {
      id: 1,
      type: 'users',
      attributes: {
        name: 'test1',
        lastname: 'data1',
        avatar: '/mock.data.com/test1.png',
        email: 'test1@testdata.com',
        job: 'tester',
        phone: '+123456789'
      }
    },
    {
      id: 2,
      type: 'users',
      attributes: {
        name: 'test2',
        lastname: 'data2',
        avatar: '/mock.data.com/test2.png',
        email: 'test2@testdata.com',
        job: 'testeador',
        phone: '+124681012'
      }
    },
    {
      id: 3,
      type: 'users',
      attributes: {
        name: 'test3',
        lastname: 'data3',
        avatar: '/mock.data.com/test3.png',
        email: 'test3@testdata.com',
        job: 'testeando',
        phone: '+1135791113'
      }
    },
    
  ],
};

export default MockData;