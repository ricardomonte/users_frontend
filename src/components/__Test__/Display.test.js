import React from 'react';
import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import MockData from '../../../tools/MockData';
import Display from '../Display';

test('Content from Display component matches snapshot', () => {
  const content = render(
    <Display users={MockData} />,
  );
  expect(content.container).toMatchSnapshot();
});