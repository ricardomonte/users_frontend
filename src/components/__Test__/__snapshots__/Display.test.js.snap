// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content from Display component matches snapshot 1`] = `
<div>
  <header
    class="mdc-top-app-bar"
  >
    <div
      class="mdc-top-app-bar__row"
    >
      <section
        class="mdc-top-app-bar__section mdc-top-app-bar__section--align-start"
      >
        <span
          class="mdc-top-app-bar__title"
        >
          My Important Contacts
        </span>
      </section>
    </div>
  </header>
  <div
    class="w40"
  >
    <div
      class="mdc-card mdc-card--outlined space_card"
    >
      <div
        class="w100 sticky"
      >
        <div
          class="mdc-text-field w100 mdc-text-field--with-trailing-icon"
        >
          <input
            class="mdc-text-field__input"
            id=""
            value=""
          />
          <label
            class="mdc-floating-label"
            for=""
          >
            Search User
          </label>
          <div
            class="mdc-line-ripple"
          />
          <i
            class="material-icons mdc-text-field__icon"
            tabindex="-1"
          >
            search
          </i>
        </div>
        <div
          class="mdc-text-field-helper-line"
        >
          <p
            aria-hidden="false"
            class="mdc-text-field-helper-text"
          >
            ie. Max, Stainer, Samantha
          </p>
        </div>
      </div>
      <div
        class="mdc-card__primary-action  cardContent mdc-elevation--z4"
      >
        <div
          class="mdc-card__media img mdc-card__media--square"
          style="background-image: url(http://localhost:4000/mock.data.com/test1.png);"
        />
        <div
          class="content_size info_card"
        >
          <div>
            <h6
              class="mdc-typography mdc-typography--headline6"
            >
              test1 data1
            </h6>
            <h6
              class="mdc-typography mdc-typography--subtitle1"
            >
              tester
            </h6>
          </div>
          <div
            class="mdc-card__action-icons"
          >
            <i
              class="material-icons mdc-card__action mdc-card__action--icon"
            >
              expand_more
            </i>
          </div>
        </div>
      </div>
      <div
        class="mdc-card__primary-action  cardContent mdc-elevation--z4"
      >
        <div
          class="mdc-card__media img mdc-card__media--square"
          style="background-image: url(http://localhost:4000/mock.data.com/test2.png);"
        />
        <div
          class="content_size info_card"
        >
          <div>
            <h6
              class="mdc-typography mdc-typography--headline6"
            >
              test2 data2
            </h6>
            <h6
              class="mdc-typography mdc-typography--subtitle1"
            >
              testeador
            </h6>
          </div>
          <div
            class="mdc-card__action-icons"
          >
            <i
              class="material-icons mdc-card__action mdc-card__action--icon"
            >
              expand_more
            </i>
          </div>
        </div>
      </div>
      <div
        class="mdc-card__primary-action  cardContent mdc-elevation--z4"
      >
        <div
          class="mdc-card__media img mdc-card__media--square"
          style="background-image: url(http://localhost:4000/mock.data.com/test3.png);"
        />
        <div
          class="content_size info_card"
        >
          <div>
            <h6
              class="mdc-typography mdc-typography--headline6"
            >
              test3 data3
            </h6>
            <h6
              class="mdc-typography mdc-typography--subtitle1"
            >
              testeando
            </h6>
          </div>
          <div
            class="mdc-card__action-icons"
          >
            <i
              class="material-icons mdc-card__action mdc-card__action--icon"
            >
              expand_more
            </i>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
